### PR TITLE
Issue #243 (CI) - Chunk up the definition fetching

### PR DESCRIPTION
**Pull request by gianpaj**
https://github.com/clearlydefined/website/issues/243

The `package-lock.json` from the `website` (this) project can be used to test this. It has over 1000 dependencies.

### CODE REVIEW

File: src/components/PageDefinitions.js [hunk 1]
```diff
@@ -249,10 +249,15 @@ class PageDefinitions extends AbstractPageDefinitions {
...
     if (listSpec.sortBy || listSpec.filter) this.setState({ sequence: this.state.sequence + 1 })
 
-    const toAdd = listSpec.coordinates.map(component => EntitySpec.validateAndCreate(component)).filter(e => e)
+    const toAdd = listSpec.coordinates.map(component => EntitySpec.validateAndCreate(component))
     dispatch(uiBrowseUpdateList({ addAll: toAdd }))
+    const chunk = 100
+    let temparray
     const missingDefinitions = toAdd.map(spec => spec.toPath()).filter(path => !definitions.entries[path])
-    await dispatch(getDefinitionsAction(token, missingDefinitions))
+    for (let i = 0, j = missingDefinitions.length; i < j; i += chunk) {
```
Comment (storrisi) [inline src/components/PageDefinitions.js]: Not so sure about using a `for` statement.
Maybe a Promise.all() could be a better elegant way?

File: src/components/PageDefinitions.js [hunk 2]
```diff
@@ -249,10 +249,15 @@ class PageDefinitions extends AbstractPageDefinitions {
...
+      temparray = missingDefinitions.slice(i, i + chunk)
+      await dispatch(getDefinitionsAction(token, temparray))
```
Comment (storrisi) [inline src/components/PageDefinitions.js]: What if the `getDefinitionsAction` receive a large array from other point of the App?
Should an implementation directly on the method could be more useful?

Comment (gianpaj) [reply src/components/PageDefinitions.js]: I would say this is unlikely but sounds fair. I'll do that

File: src/components/PageDefinitions.js (hunk 1 above)
Comment (gianpaj) [reply src/components/PageDefinitions.js]: Yes. I was thinking about this as I was just writing the PR. I'll do that

**Comment by gianpaj** [comment]
I've tested using dev's API **before** it took 26.5 seconds

clearlydefined


and **after** it took 33 seconds now
clearlydefined

Note: this is only one test each.

We could try to remove the `Promise.all()` as you suggested Simo and trigger the dispatch actions (start, success/error) mutiple times instead of waiting for all the Promises to finish.

**Comment by storrisi** [comment]
Surely this implementation doesn't solve the problem, so we have to try other routes.

**Pull request merged**